Client/Impl/Worker/JobWorker.cs

```diff
@@ -77,10 +86,95 @@ internal void Open()
         {
             isRunning = true;
             var cancellationToken = source.Token;
-            var jobWorkerSignal = new JobWorkerSignal();
 
-            StartPollerThread(jobWorkerSignal, cancellationToken);
-            StartHandlerThreads(jobWorkerSignal, cancellationToken);
+            activateJobsCommand = jobWorkerBuilder.Command;
+            maxJobsActive = jobWorkerBuilder.Command.Request.MaxJobsToActivate;
+
+            // Create options for blocks
+            var bufferOptions = new DataflowBlockOptions
+            {
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            var executionOptions = new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = jobWorkerBuilder.ThreadCount,
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            // Buffer for polled jobs
+            var input = new BufferBlock<IJob>(bufferOptions);
+
+            // Transformblock to process polled jobs
+            var transformer = new TransformBlock<IJob, IJob>(async activatedJob =>
+                {
+                    try
+                    {
+                        await jobHandler(jobClient, activatedJob);
+                        await TryToAutoCompleteJob(activatedJob);
+                    }
+                    catch (Exception exception)
+                    {
+                        await FailActivatedJob(activatedJob, cancellationToken, exception);
+                    }
+                    finally
+                    {
+                        jobClient.Reset();
+                    }
+
+                    return activatedJob;
+                },
+                executionOptions);
+
+            // Action block to finalize handled tasks
+            var output = new ActionBlock<IJob>(activatedJob => { currentJobsActive--; },
```

Comment:
We could also use a Batchblock to activate the next polling after a certain threshold. What do you think?

Reply:
Not sure. I tried to find a way where consumed jobs trigger the poll loop again, but had no idea how to do that. But I think the one loop which Task.Delays most of the time is absolutely no performance problem.

Reply:
I think you're right